common/sensor/sdr.c

```diff
@@ -50,36 +49,127 @@ bool SDR_RSV_ID_check(uint16_t ID)
 	return (RSV_ID == ID) ? 1 : 0;
 }
 
-uint8_t SDR_init(void)
+__weak void pal_fix_full_sdr_table(void)
+{
+	return;
+}
+
+uint8_t get_sdr_index(uint8_t sensor_num)
+{
+	uint8_t i, j;
+	for (i = 0; i < SENSOR_NUM_MAX; i++) {
+		for (j = 0; j < SDR_NUM; ++j) {
+			if (sensor_num == full_sdr_table[j].sensor_num) {
+				return j;
+			} else if (i == SDR_NUM) {
+				return SENSOR_NUM_MAX;
+			}
+		}
+	}
+
+	return SENSOR_NUM_MAX;
+}
+
+void add_full_sdr_table(SDR_Full_sensor add_item)
+{
+	if (get_sdr_index(add_item.sensor_num) != SENSOR_NUM_MAX) {
+		printf("add sensor num [%x] is already exists\n", add_item.sensor_num);
```

Comment:
"add sensor num [%x] already exists\n"

Reply:
We have fixed this and done the forced update.
Please review it. Thanks.